hw05_parallel_execution/run.go

```diff
@@ -2,14 +2,51 @@ package hw05parallelexecution
 
 import (
 	"errors"
+	"sync"
+	"sync/atomic"
 )
 
 var ErrErrorsLimitExceeded = errors.New("errors limit exceeded")
 
 type Task func() error
 
-// Run starts tasks in n goroutines and stops its work when receiving m errors from tasks.
-func Run(tasks []Task, n, m int) error {
-	// Place your code here.
+func Run(tasks []Task, runnersLimit, errorsLimit int) error {
+	var errors int32
+	wg := sync.WaitGroup{}
+	ch := tasksToChannel(tasks)
+
+	for runner := 0; runner < runnersLimit; runner++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			startRunner(&ch, &errors, int32(errorsLimit))
+		}()
+	}
+
+	wg.Wait()
+	if errors >= int32(errorsLimit) {
+		return ErrErrorsLimitExceeded
+	}
 	return nil
 }
+
+func tasksToChannel(tasks []Task) chan Task {
+	ch := make(chan Task, len(tasks))
+	defer close(ch)
+	for _, task := range tasks {
+		ch <- task
+	}
+	return ch
+}
+
+func startRunner(channel *chan Task, errors *int32, errorsLimit int32) {
```

Comment:
вам нет необходимости использовать указатель канала, т.к. канал сам по себе уже является указателем. 
```suggestion
func startRunner(сhannel chan Task, errors *int32, errorsLimit int32) {
```